src/chem/mopitt_co_nc2ioda.py

```diff
@@ -0,0 +1,223 @@
+#!/usr/bin/env python3
+
+#
+# (C) Copyright 2020 UCAR
+#
+# This software is licensed under the terms of the Apache Licence Version 2.0
+# which can be obtained at http://www.apache.org/licenses/LICENSE-2.0.
+#
+
+import sys
+import argparse
+import netCDF4 as nc
+import numpy as np
+from datetime import datetime, timedelta
+import os
+from pathlib import Path
+
+IODA_CONV_PATH = Path(__file__).parent/"@SCRIPT_LIB_PATH@"
+if not IODA_CONV_PATH.is_dir():
+    IODA_CONV_PATH = Path(__file__).parent/'..'/'lib-python'
+sys.path.append(str(IODA_CONV_PATH.resolve()))
+
+import ioda_conv_engines as iconv
+from collections import defaultdict, OrderedDict
+from orddicts import DefaultOrderedDict
+
+locationKeyList = [
+    ("latitude", "float"),
+    ("longitude", "float"),
+    ("datetime", "string"),
+]
+
+obsvars = {
+    'carbonmonoxide_total_column': 'carbon_monoxide_in_total_column',
+}
+
+AttrData = {
+    'converter': os.path.basename(__file__),
+    'nvars': np.int32(len(obsvars)),
+}
+
+DimDict = {
+}
+
+VarDims = {
+    'carbon_monoxide_in_total_column': ['nlocs'],
+}
+
+class mopitt(object):
+    def __init__(self, filenames):
+        self.filenames = filenames
+        self.varDict = defaultdict(lambda: defaultdict(dict))
+        self.outdata = defaultdict(lambda: DefaultOrderedDict(OrderedDict))
+        self.varAttrs = DefaultOrderedDict(lambda: DefaultOrderedDict(dict))
+        self._read()
+
+    def _read(self):
+        # set up variable names for IODA
+        for iodavar in ['carbon_monoxide_in_total_column']:
+            self.varDict[iodavar]['valKey'] = iodavar, iconv.OvalName()
+            self.varDict[iodavar]['errKey'] = iodavar, iconv.OerrName()
+            self.varDict[iodavar]['qcKey'] = iodavar, iconv.OqcName()
+            self.varAttrs[iodavar, iconv.OvalName()]['coordinates'] = 'longitude latitude'
+            self.varAttrs[iodavar, iconv.OerrName()]['coordinates'] = 'longitude latitude'
+            self.varAttrs[iodavar, iconv.OqcName()]['coordinates'] = 'longitude latitude'
+            self.varAttrs[iodavar, iconv.OvalName()]['units'] = 'mol m-2'
+            self.varAttrs[iodavar, iconv.OerrName()]['units'] = 'mol m-2'
+            self.varAttrs[iodavar, iconv.OqcName()]['units'] = 'unitless'
+        # loop through input filenames
+        first = True
+        for f in self.filenames:
+
+            ncd = nc.Dataset(f, 'r')
+
+            #AttrData['date_time_string']
+            StartDateTime = str(ncd.groups['HDFEOS'].groups['ADDITIONAL'].groups['FILE_ATTRIBUTES'].StartDateTime)
+            AttrData['sensor'] = 'MOPITT'
+            AttrData['platform'] = 'Terra'
```

Comment:
Is this only on Terra? Is there an attribute in the original file that contains this?

Reply:
Yes it's only Terra and unfortunately no attributes in the files